File: apple/Sources/ReactNativeWatchOS/Modifiers/LineSpacing.swift
import SwiftUI

/// SwiftUI `.lineSpacing(_:)`. Sets the vertical distance between lines of
/// text within the view. An absent `value` leaves the view unchanged.
enum RNWLineSpacingModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("lineSpacing") { view, params, _ in
            guard let spacing = params.cgFloat("value") else { return view }
            return AnyView(view.lineSpacing(spacing))
        }
    }
}

File: src/modifiers/lineSpacing.ts
import { createModifier } from './createModifier';

export interface LineSpacingParams {
  /// Spacing in points between lines of text.
  value?: number;
}

/// SwiftUI `.lineSpacing(_:)`. Sets the vertical distance between lines of
/// text within the view. Pass a number directly or an object.
export function lineSpacing(params?: LineSpacingParams | number) {
  if (typeof params === 'number') {
    return createModifier('lineSpacing', { value: params });
  }
  return createModifier('lineSpacing', params);
}
